R/dvhydrograph-render.R

```diff
@@ -3,199 +3,291 @@ dvhydrographPlot <- function(data) {
+
+  #Validate Basic Plot Data
+  if(is.null(primaryPresentTS)){
+    return(NULL)
+  }
+
+  #Get Estimated / Non-Estimated Edges
+  if(!is.null(dvData[['stat1TimeSeries']]) && !is.null(dvData[['stat1TimeSeriesEst']])){
+    dvData[['estimated1Edges']] <- getEstimatedEdges(dvData[['stat1TimeSeries']][['points']], dvData[['stat1TimeSeriesEst']][['points']])
+  }
+
+  if(!is.null(dvData[['stat2TimeSeries']]) && !is.null(dvData[['stat2TimeSeriesEst']])){
+    dvData[['estimated2Edges']] <- getEstimatedEdges(dvData[['stat2TimeSeries']][['points']], dvData[['stat2TimeSeriesEst']][['points']])
+  }
+
+  if(!is.null(dvData[['stat3TimeSeries']]) && !is.null(dvData[['stat3TimeSeriesEst']])){
+    dvData[['estimated3Edges']] <- getEstimatedEdges(dvData[['stat3TimeSeries']][['points']], dvData[['stat3TimeSeries']][['points']])
+  }
+
+  if(!is.null(dvData[['comparisonTimeSeries']]) && !is.null(dvData[['comparisonTimeSeriesEst']])){
+    dvData[['comparisonEdges']] <- getEstimatedEdges(dvData[['comparisonTimeSeries']][['points']], dvData[['comparisonTimeSeriesEst']][['points']])
+  }
+
+  #Get Additional Plot Data
+  dvData[['gw_level']] <- parseDVGroundWaterLevels(reportObject)
+  dvData[['meas_Q']] <- parseDVFieldVisitMeasurements(reportObject)
+  dvData <- append(dvData, parseDVMinMaxIVs(reportObject, timezone, dvData[[primaryPresentTS]][['type']], invertedFlag, excludeMinMaxFlag, excludeZeroNegativeFlag))
+  dvData <- append(dvData, parseDVApprovals(dvData[[primaryPresentTS]], timezone))
+  logAxis <- isLogged(dvData[[primaryPresentTS]][['points']], dvData[[primaryPresentTS]][['isVolumetricFlow']], excludeZeroNegativeFlag)
+  yLabel <- paste0(dvData[[primaryPresentTS]][['type']], ", ", dvData[[primaryPresentTS]][['units']])
+
+  #Do plotting
+  plot_object <- gsplot(ylog = logAxis, yaxs = 'i') %>%
       grid(nx = 0, ny = NULL, equilogs = FALSE, lty = 3, col = "gray") %>%
       axis(1, at = plotDates, labels = format(plotDates, "%b\n%d"), padj = 0.5) %>%
-      axis(2, reverse = isInverted) %>%
-      view(xlim = c(startDate, endDate)) %>%
-      title(
-        ylab = paste0(data$firstDownChain$type, ", ", data$firstDownChain$units),
-        line = 3
-      )
-
-    plot_object <-
-      XAxisLabelStyle(plot_object, startDate, endDate, data$reportMetadata$timezone, plotDates)
+      axis(2, reverse = invertedFlag, las=0) %>%
+      view(xlim = c(startDate, endDate))
     
-    # for non-approval-bar objects
-    for (i in grep("^appr_", names(dvData), invert = TRUE)) {
-      dvConfig <- getDVHydrographPlotConfig(dvData[i], dvInfo)
-      for (j in names(dvConfig)) {
-        dvConfig[[j]] <- extendStep(dvConfig[[j]])
-        plot_object <- do.call(names(dvConfig[j]), append(list(object = plot_object), dvConfig[[j]]))
-      }
+  plot_object <-
+    XAxisLabelStyle(plot_object, startDate, endDate, timezone, plotDates)
+
+  # for non-approval-bar objects
+  for (i in grep("^appr_", names(dvData), invert = TRUE)) {
```

Comment:
I haven't gotten this far in my UV Hydro refactor but anticipated coming up to the same thing. Do you think it's worth while not to collect all of the data in a dvData list? This loop, for example, would be replaced by individual calls for each approval data element.

It's more code, but I get nervous doing string matches against env items to find things. There is a lot of potential for unseen side effects (eg, someone adds a env var latter that matches unintentionally).